examples/src/bin/composite_template.rs

```diff
@@ -0,0 +1,114 @@
+//! # Composite Template Example
+//!
+//! This sample demonstrates how to create a widget using GTK's composite templates.
+
+#[macro_use]
+extern crate glib;
+#[macro_use]
+extern crate gtk;
+
+use gio::{prelude::ApplicationExtManual, ApplicationExt};
+use glib::subclass::prelude::*;
+use glib::{Cast, StaticType};
+use gtk::WidgetExt;
+
+mod imp {
+    use super::*;
+    use glib::subclass;
+    use gtk::subclass::prelude::*;
+    use gtk::subclass::widget::*;
+    use gtk::WidgetExt;
+
+    /// The private struct, which can hold widgets and other data.
+    #[derive(Debug, CompositeTemplate)]
+    pub struct ExApplicationWindow {
+        // The #[template_child] attribute tells the CompositeTemplate macro
+        // that a field is meant to be a child within the template.
+        #[template_child(id = "headerbar")]
+        pub headerbar: TemplateChild<gtk::HeaderBar>,
+        #[template_child(id = "label")]
+        pub label: TemplateChild<gtk::Label>,
+        #[template_child(id = "subtitle_label")]
+        pub subtitle: TemplateChild<gtk::Label>,
+    }
+
+    impl ObjectSubclass for ExApplicationWindow {
+        const NAME: &'static str = "ExApplicationWindow";
+        type Type = super::ExApplicationWindow;
+        type ParentType = gtk::ApplicationWindow;
+        type Instance = subclass::simple::InstanceStruct<Self>;
+        type Class = subclass::simple::ClassStruct<Self>;
+
+        glib_object_subclass!();
+
+        fn new() -> Self {
+            Self {
+                headerbar: TemplateChild::default(),
+                label: TemplateChild::default(),
+                subtitle: TemplateChild::default(),
+            }
+        }
+
+        // Within class_init() you must set the template
+        // and bind it's children. The CompositeTemplate
+        // derive macro provides a convenience function
+        // bind_template_children() to bind all children
+        // at once.
+        fn class_init(klass: &mut Self::Class) {
+            let template = include_bytes!("composite_template.ui");
+            klass.set_template(template);
+            Self::bind_template_children(klass);
+        }
+    }
+
+    impl ObjectImpl for ExApplicationWindow {
+        fn constructed(&self, obj: &Self::Type) {
+            obj.init_template();
+            obj.init_label();
+            self.parent_constructed(obj);
+        }
+    }
+
+    impl WidgetImpl for ExApplicationWindow {}
+    impl WindowImpl for ExApplicationWindow {}
+    impl ApplicationWindowImpl for ExApplicationWindow {}
+}
+
+glib_wrapper! {
```

Comment:
```suggestion
glib::glib_wrapper! {
```

Reply:
I've changed this in a different way - mind taking a look?

Reply:
Seems OK to me too